zebra-network/src/peer/client/tests.rs

```diff
@@ -1,3 +1,153 @@
-//! Tests for the [`Client`] part of peer connections
+//! Tests for the [`Client`] part of peer connections, and some test utilities for mocking
+//! [`Client`] instances.
 
 mod vectors;
+
+use futures::channel::{mpsc, oneshot};
+
+use crate::{
+    peer::{error::SharedPeerError, CancelHeartbeatTask, Client, ClientRequest, ErrorSlot},
+    protocol::external::types::Version,
+};
+
+/// A handle to a mocked [`Client`] instance.
+pub struct MockedClientHandle {
+    request_receiver: Option<mpsc::Receiver<ClientRequest>>,
+    shutdown_receiver: Option<oneshot::Receiver<CancelHeartbeatTask>>,
+    error_slot: ErrorSlot,
+    version: Version,
+}
+
+impl MockedClientHandle {
+    /// Create a new mocked [`Client`] instance, returning it together with a handle to track it.
```

Comment:
```suggestion
    /// Create a new [`Client`] instance with mocked channels, returning it together with a handle to track it.
```

Reply:
This constructor has been removed and replaced with the `TestClientBuilder`.